### ISSUE TITLE
bad_channel_ids gives a Matrix dimensions must agree.Error using /

### ISSUE THREAD

**Issue by gmaggi** [SpikeInterface/spiketoolkit#471]
Hello,

There is a dataset that runs well if `bad_channel_ids = []`, nevertheless,`bad_channel_ids = [191]` throws the following (this is the last part of kilosort2.log):
...
Finished splitting. Found 55 splits, checked 252/252 clusters, nccg 80
Found 0 splits, checked 1/252 clusters, nccg 0
Found 0 splits, checked 1/252 clusters, nccg 0
Found 4 splits, checked 101/256 clusters, nccg 30
Found 4 splits, checked 101/256 clusters, nccg 30
Found 7 splits, checked 201/259 clusters, nccg 54
Found 7 splits, checked 201/259 clusters, nccg 54
Finished splitting. Found 8 splits, checked 260/260 clusters, nccg 78
Finished splitting. Found 8 splits, checked 260/260 clusters, nccg 78
--------------------------------------------------------------------------------Error using /
Matrix dimensions must agree.Error using /
Matrix dimensions must agree.

Any idea what is causing this error?

The user is using the following values:

`bad_channel_ids: [191]
freq_min: 300
freq_max: 60000
reference "average"`

Thanks!

**Comment by alejoe91** [comment]
Hi @gmaggi 

Strange that kilosort runs fine almost until the end.. It might be a problem with the way `remove_bad_channels` and `common_reference` interact with each other. Can you try to run KS removing bad_channels, but not applying CAR? At least we can see if that is the case

**Comment by gmaggi** [comment]
Hi @alejoe91 ,

I got some feedback from my colleague. It turns out that if the `common_reference` method is not executed, preprocessing works fine with  `bad_channel_ids= [...]`

Feel free to close this tissue.
Thanks,
Giuliano.

**Comment by MarineChap** [comment]
@gmaggi  Hey not sure why you would want this to be closed ? Hanna needs it to work together.
@alejoe91 you think there is no way to fix this problem ? 

Thanks

**Comment by gmaggi** [comment]
Sorry for my early closing request. Maybe @alejoe91 has further inputs.

**Comment by alejoe91** [comment]
> Hi @alejoe91 ,
> 
> I got some feedback from my colleague. It turns out that if the `common_reference` method is not executed, preprocessing works fine with `bad_channel_ids= [...]`
> 
> Feel free to close this tissue.
> Thanks,
> Giuliano.

Can you summarize again the problem? I got a bit confused by this message

**Comment by gmaggi** [comment]
Hi @alejoe91 

The problem (Matrix dimensions must agree.Error using /) is when using preprocessing tools:

- **Case it works fine:** 

1. outBadChannels = st.preprocessing.remove_bad_channels(recoding, bad_channel_ids=[191], defaults)

2. outBandPass = st.preprocessing.bandpass_filter(outBadChannels, freq_min=300, freq_max=60000, defaults)

- **Case it gives troubles:**

As before, plus:

3. st.preprocessing.common_reference(outBandPass, reference= 'average')

Thanks,
Giuliano.

**Comment by alejoe91** [comment]
I see. Did you try to disable Kilosort's CAR? I don't see why this would fail..maybe the global CAR messes up the signals a bit? If Kilosort runs, then the recordings should be "correct"

**Comment by MarineChap** [comment]
Hello @alejoe91
I just checked with Hanna but we are always disabling the Kilosort CAR option in our lab. 
Do you think we should also open an issue on the kilosort repo ?

**Comment by alejoe91** [comment]
Can you check the signals that you are ingesting into kilosort? Also if you don't use the bad_channels does it work?

**Comment by MarineChap** [comment]
What do you want us to check on the signals exactly ? 
and yes without bad_channels it works. that's really the combination of CAR and bad_channels. 

This afternoon we will look if we see any majors differences between the output of: 
- CAR followed by bad_channels removal
- bad_channels removal followed by CAR
- CAR alone

Keep you posted.

**Comment by MarineChap** [comment]
Comparison done by Hanna: 

```
Unprocessed data:
Num. channels = 384
Sampling frequency = 30000.0 Hz
Num. timepoints = 5000
Stdev. on 192th channel = 43.93719076317921
Location of 192th electrode = [   0. 5760.]

CAR only:
Num. channels = 384
Sampling frequency = 30000.0 Hz
Num. timepoints = 5000
Stdev. on 192th channel = 15.968848513277342
Location of 192th electrode = [   0. 5760.]

CAR plus remove channel 191:
Num. channels = 383
Sampling frequency = 30000.0 Hz
Num. timepoints = 5000
Stdev. on 192th channel = 15.968848513277342
Location of 192th electrode = [   0. 5760.]

Remove channel 191 followed by CAR:
Num. channels = 383
Sampling frequency = 30000.0 Hz
Num. timepoints = 5000
Stdev. on 192th channel = 15.920968110011401
```
Seems all good. Do you see something else to check ?

**Comment by alejoe91** [comment]
Not really honestly. Once kilosort is run, it works on the binary file created, so KS "doesn't know" that there is a missing channel!

**Comment by MarineChap** [comment]
yeap I am also puzzled by this.  I just tested with Kilosort3 instead of Kilosort2 and it is working.

**Comment by MarineChap** [comment]
Hello, with the last release of spiketoolkit (0.7.5), this bug disappeared.  So we can close this issue!
(I cannot do myself apparently as I am not the person who opened it or maintainers)

**Comment by alejoe91** [comment]
Closing then :)
